### DISCUSSION TITLE
Is it posible to export with --convert-to-jpeg and discard the original RAW?

### DISCUSSION THREAD

**Discussion by odedia** [RhetTbull/osxphotos#1509] [General]
I've seen there's a parameter called `--skip-raw` but docs say this is specifically for RAW+JPEG combos. Is there a way to tell osxphotos to export converted RAW files as JPEGs but not include the RAW file in the output?
Closest thing I've found is `--skip-original-if-edited` but that only applies to edited photos (which is great!), but it wouldn't apply for plain-old RAW files that were never edited.

**Comment by odedia** [comment]
Seems like it already does this by default?
I don't see any .NEF files in my output folder with the below combo:

```
--edited-suffix "" --update --cleanup --skip-raw --skip-original-if-edited --convert-to-jpeg
```

**Reply by RhetTbull** [reply]
The `--skip-raw` causes the RAW component of the RAW+JPEG pair to be skipped but stand-alone RAW images are still exported.

The `--convert-to-jpeg` causes the stand-alone RAW images to be converted to JPEG but the RAW+JPEG pairs are left intact.

Used together, you'll get no RAW images in the output.  Here's sample output from each option:

`osxphotos export`

All assets exported.

```
DSC03584.dng <-- RAW image
IMG_1994.cr2 <-- RAW + JPEG
IMG_1994.JPG
```

`osxphotos export --skip-raw`

Stand-alone RAW image is still exported but the RAW image from a RAW+JPEG pair is not exported.

```
DSC03584.dng
IMG_1994.JPG
```

`osxphotos export --convert-to-jpeg`

Stand-alone RAW is converted to JPEG but RAW+JPEG pairs are left as-is.

```
DSC03584.jpeg <-- RAW converted to JPEG
IMG_1994.cr2  <-- RAW pair is left intact
IMG_1994.JPG
```

`osxphotos export --convert-to-jpeg --skip-raw`

Stand-alone RAW converted to JPEG, RAW component of RAW+JPEG pair is not exported.

```
DSC03584.jpeg <-- converted to JPEG
IMG_1994.JPG <-- RAW component of RAW+JPEG pair is skipped
```

**Comment by odedia** [comment]
This is exactly like I felt it should work, you're like a mind-reader 🙌
Thank you.

**Comment by odedia** [comment]
This doesn't seem to work as I would expect, I'm sure I'm missing something.
I'll use `_DSC6704-Enhanced-NR` as my sample image.
In Apple Photos, I have it as a RAW+JPEG combo. If I export originals using Apple Photos, I get:
```
_DSC6704-Enhanced-NR.dng
_DSC6704-Enhanced-NR.jpg
```

When I run OSXPhotos with the below toml:

```
[export]
cleanup = true
convert_to_jpeg = false
db = "/Volumes/ExternalData/Users/odedia/Pictures/Photo Library.photoslibrary"
directory = "{created.year}/{created.year}-{created.mm}-{created.dd}"
download_missing = true
export_aae = false
keyword_template = [ "{label}", "{keyword}", "{person}", "{album?album:{folder_album}}", "{favorite?FavoriteInApplePhotos,}",]
replace_keywords = true
retry = 0
sidecar = [ "xmp",]
skip_edited = true
tmpdir = "/Volumes/ExternalData/Users/odedia/Pictures/OSXPhotos/TempDir"
update = true
use_photokit = true
verbose = 0
```

I get the following in the output folder:
```
_DSC6710-Enhanced-NR (1).dng
_DSC6710-Enhanced-NR (1).jpg
_DSC6710-Enhanced-NR (1).jpg.xmp
_DSC6710-Enhanced-NR.jpg
_DSC6710-Enhanced-NR.jpg.xmp
```

When I run OSXPhotos with the following toml:
```
[export]
cleanup = true
convert_to_jpeg = true
db = "/Volumes/ExternalData/Users/odedia/Pictures/Photo Library.photoslibrary"
directory = "{created.year}/{created.year}-{created.mm}-{created.dd}"
download_missing = true
edited_suffix = ""
keyword_template = [ "{label}", "{keyword}", "{person}", "{album?album:{folder_album}}", "{favorite?FavoriteInApplePhotos,}",]
replace_keywords = true
retry = 0
sidecar = [ "xmp",]
skip_original_if_edited = true
skip_raw = true
tmpdir = "/Volumes/ExternalData/Users/odedia/Pictures/OSXPhotos/TempDir"
update = true
use_photokit = true
```

I get the following in the output folder:
```
_DSC6710-Enhanced-NR (1).jpg
_DSC6710-Enhanced-NR (1).jpg.xmp
_DSC6710-Enhanced-NR (2).jpg
_DSC6710-Enhanced-NR (2).jpg.xmp
```

Not sure why it behaves like this.

**Reply by RhetTbull** [reply]
I'll take a look. `--convert-to-jpeg` should not convert the RAW component and I confirmed on a test image that it does not. Will try with your full TOML later incase there's something else going on.

```
--convert-to-jpeg               Convert all non-JPEG images (e.g. RAW, HEIC, PNG, etc) to JPEG upon export.
                                Note: does not convert the RAW component of a RAW+JPEG pair as the
                                associated JPEG image will be exported. You can use --skip-raw to skip
                                exporting the associated RAW image of a RAW+JPEG pair. See also --jpeg-
                                quality and --jpeg-ext. Only works if your Mac has a GPU (thus may not work
                                on virtual machines).
```
